listenbrainz/listenstore/timescale_listenstore.py

```diff
@@ -995,46 +964,53 @@ def delete(self, musicbrainz_id, user_id):
 
         Raises: Exception if unable to delete the user in 5 retries
         """
-
-        self.set_empty_cache_values_for_user(musicbrainz_id, user_id)
-        args = {'user_id': user_id}
-        query = "DELETE FROM listen WHERE user_id = :user_id"
-
+        query = """
+            UPDATE listen_helper SET count = 0 WHERE user_id = :user_id;
+            DELETE FROM listen WHERE user_id = :user_id;
+        """
+        cache.set(REDIS_USER_TIMESTAMPS + str(user_id), "0,0", 0)
         try:
             with timescale.engine.connect() as connection:
-                connection.execute(sqlalchemy.text(query), args)
+                connection.execute(sqlalchemy.text(query), user_id=user_id)
         except psycopg2.OperationalError as e:
             self.log.error("Cannot delete listens for user: %s" % str(e))
             raise
 
-    def delete_listen(self, listened_at: int, user_name: str, user_id: int, recording_msid: str):
+    def delete_listen(self, listened_at: int, user_id: int, recording_msid: str):
         """ Delete a particular listen for user with specified MusicBrainz ID.
         Args:
             listened_at: The timestamp of the listen
-            user_name: the username of the user
-            user_id: the user id of the user
+            user_id: the listenbrainz row id of the user
             recording_msid: the MessyBrainz ID of the recording
         Raises: TimescaleListenStoreException if unable to delete the listen
         """
-
-        args = {
-            'listened_at': listened_at,
-            'user_id': user_id,
-            'recording_msid': recording_msid
-        }
-        query = """DELETE FROM listen
-                    WHERE listened_at = :listened_at
-                      AND user_id = :user_id
-                      AND data -> 'track_metadata' -> 'additional_info' ->> 'recording_msid' = :recording_msid """
-
+        logging.getLogger('sqlalchemy.engine').setLevel(logging.INFO)
```

Comment:
The timestamps would not be correct if the first or the last listen were to be deleted via this function.